### PR TITLE
Soen117 patient dashboard

### PR DESCRIPTION
**Pull request by KelvinTrang**
# Description

Implements/fixes [SOEN-117](https://soen390team8.atlassian.net/browse/SOEN-117)

- Fix bug for status charts in dashboard and forms to fill
- Added a line to tell the patient if he has filled today's status or not

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- [X] Test Backend and frontend
- [ ] Test B
- [ ] Test C

# Screenshots
image

# Comments
The message is in red but if you have any suggestion please tell me!

### CODE REVIEW

**Review by silentswordxv — APPROVED** [review]
Good work!

**Review by stevenvuong00 — APPROVED** [review]
great job bingbong! looks good

File: frontend/pages/index.tsx [hunk 1]
```diff
@@ -11,15 +11,28 @@ export const getServerSideProps: GetServerSideProps = async context => {
...
+    if (role === USER_ROLES.patient) {
+        let response = await fetch(serverURL + "/status/getAllStatusChart/" + userId);
+        data = await response.json();
+
+        // status filled for the day?
+        response = await fetch(serverURL + "/status/getAllStatus/" + userId);
+        statusOfTheDay = await response.json();
+
+        statusFilled = statusOfTheDay[0].StatusTime.substring(0, 10) === today;
+    }
```
Comment (NicholasYip) [inline frontend/pages/index.tsx:29]: you could combine all these promises with `Promise.all([promise1, promise2, promise3])`

File: frontend/pages/patient/patient-symptoms-daily.tsx [hunk 2]
```diff
@@ -19,13 +19,18 @@ export async function getServerSideProps(context: NextPageContext) {
...
     if (session?.user.Role === USER_ROLES.patient) {
         try {
             let response: any = await fetch(serverURL + "/patients/getRequiredDetails/" + userId);
             requiredDetails = getFieldsForPatient(await response.json());
 
             response = await fetch(serverURL + "/status/getAllStatus/" + userId);
             pastConditions = await response.json();
+
+            response  = await fetch(serverURL + "/status/getAllStatusChart/" + userId);
+            statusChartData = await response.json();
```
Comment (NicholasYip) [inline frontend/pages/patient/patient-symptoms-daily.tsx]: you could combine all these promises here also

**Review by NicholasYip — APPROVED** [review]
small comment, rest looks good

**Pull request merged**